### ISSUE TITLE
Allow specification of `-modify-index 0` in the consul cli

### ISSUE THREAD

**Issue by dansteen** [hashicorp/consul#3814]
### Description of the Issue (and unexpected/desired result)

The consul cli does not allow setting `-modify-index 0` (for use as described [here](https://www.consul.io/api/kv.html#cas) ).  Instead it says:

```
#> consul kv put -cas -modify-index 0 deploy_version/admin-interface/stag/historical3 '1 2 3 4 5'  
Must specify -modify-index with -cas!
```

### Reproduction steps
See above

### `consul version` for both Client and Server
Client: `v1.0.2`
Server: `v1.0.2`

### `consul info` for both Client and Server
Client:
```
agent:
        check_monitors = 0
        check_ttls = 1
        checks = 1
        services = 1
build:
        prerelease =
        revision = b55059f
        version = 1.0.2
consul:
        bootstrap = false
        known_datacenters = 1
        leader = false
        leader_addr = 10.0.253.124:8300
        server = true
raft:
        applied_index = 42139698
        commit_index = 42139698
        fsm_pending = 0
        last_contact = 17.128588ms
        last_log_index = 42139698
        last_log_term = 11837
        last_snapshot_index = 42139327
        last_snapshot_term = 11837
        latest_configuration = [{Suffrage:Voter ID:8ed7e1df-c7dc-5a8a-09c3-78187f57a638 Address:10.0.254.75:8300} {Suffrage:Voter ID:20a02ab2-0609-4730-b169-00f793f51df4 Address:10.0.253.124:8300} {Suffrage:Voter ID:cf171582-581d-43a7-9a59-a6ed71eadb67 Address:10.0.252.160:8300}]
        latest_configuration_index = 42137401
        num_peers = 2
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Follower
        term = 11837
runtime:
        arch = amd64
        cpu_count = 1
        goroutines = 504
        max_procs = 1
        os = linux
        version = go1.9.2
serf_lan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 7793
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 1159
        members = 43
        query_queue = 0
        query_time = 142
serf_wan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 64
        members = 3
        query_queue = 0
        query_time = 3

```

Server:
```
Consul cli is talking directly to the server.
```

### Operating system and Environment details
arch-linux cli
debian server

**Comment by GreatSnoopy** [comment]
This is indeed an inconsistency between the API and the CLI. It is also a very useful feature to help orchestrate some tasks (we can use this so that only the first contender that gets to create a certain initially nonexistent key will be able to execute some further steps)

**Comment by zilman** [comment]
There was a pull request #3579 that would have fixed this, but it never got merged. This issue still exists in 1.3.0.

For people googling:
`curl --request PUT localhost:8500/v1/kv/foo?cas=0`

Will only create key foo if it didn't already exist. So for example you can make a unit file that would run on all of your leaders with the above as an ExecStartPre to ensure only one leader executes some piece of logic.

**Comment by blake** [comment]
Fixed by #8331 which will be in Consul 1.9.
